core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java

```diff
@@ -54,6 +54,8 @@
     private static final String[] ALLOWED_SUFFIXES = {".yml", ".yaml", ".json", ".properties"};
     static final String[] PROPERTY_PREFIXES = {"es.", "elasticsearch."};
     static final String[] PROPERTY_DEFAULTS_PREFIXES = {"es.default.", "elasticsearch.default."};
+    // this is used to filter out things like `es.tests.portfile` which should only be accessed via system properties directly
+    static final String[] PROPERTY_DEFAULTS_AND_TEST_PREFIXES = {"es.default.", "elasticsearch.default.", "es.tests",  "elasticsearch.tests"};
```

Comment:
es.tests. <- DOT at the end
same for elasticsearch.tests

feels hacky that we have to filter out test namepace in prod code. I wonder whether, instead, es.tests.portfile should be moved to es.portfile and just be an internal undocumented setting.